libs/wingsdk/src/target-tf-aws/bucket.ts

```diff
@@ -46,20 +47,54 @@ export const BUCKET_PREFIX_OPTS: NameOptions = {
   suffix: "-",
 };
 
+/**
+ * Properties for a new tfaws.Bucket.
+ */
+export interface BucketProps extends cloud.BucketProps {
+  /**
+   * This internal property is used to override the default bucket creation
+   * and instead use an existing bucket.
+   *
+   * @internal
+   */
+  _existingBucket?: DataAwsS3Bucket;
+}
+
 /**
  * AWS implementation of `cloud.Bucket`.
  *
  * @inflight `@winglang/sdk.cloud.IBucketClient`
  */
 export class Bucket extends cloud.Bucket implements IAwsBucket {
-  private readonly bucket: S3Bucket;
-  private readonly public: boolean;
+  /**
+   * Create a new cloud.Bucket from an existing s3 bucket.
+   *
+   * @param scope scope to use for the bucket data lookup
+   * @param bucketName the name of the existing bucket to reference
+   * @returns A cloud.Bucket
+   */
+  public static fromBucketName(scope: Construct, bucketName: string): Bucket {
```

Comment:
We also need `fromBucketArn()`

Reply:
So terraform doesnt actually support data lookups based on bucket arns, only bucket names: https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/s3_bucket#argument-reference

I assume this is because Arns are not needed to identify uniqueness as S3 bucket names are globally unique.

Now we could technically still provide the API since the arn contains the bucket name, we would just parse the bucket name from the arn and look it up like that, but I wonder how useful it is to have both APIs?

Reply:
We can start with just the name of that feels okay but in my experience the idiomatic way in AWS to reference resources is ARN so all of our `from` methods should have at least an import from ARN